Packs/Cisco-umbrella/Integrations/CiscoUmbrellaEnforcement/CiscoUmbrellaEnforcement.py

```diff
@@ -0,0 +1,188 @@
+                    of queries which are then returned with paging and have a next page request built already.
+    :return: (str) with the suffix.
+    """
+    suffix = ''
+    if request:
+        suffix = request
+    if page:
+        suffix += f'page={page}&'
+    if limit:
+        suffix += f'limit={limit}&'
+    return suffix
+
+
+def create_domain_list(client: Client, response: dict) -> list:
+    """
+    :param client: Cisco Umbrella Client for the api request.
+    :param response: the first response of the domains api call.
+    :return: list of all of the domains from the api.
+    """
+    full_domains_list = []
+    while response:
+        response_data = response.get('data', [])
+        for domain in response_data:
+            full_domains_list.append(domain.get('name'))
+        response_next_page = response.get('meta', {}).get('next', '')
+        response = client.get_domains_list(response_next_page) if response_next_page else {}
+    return full_domains_list
+
+
+def domains_list_command(client: Client, args: dict) -> CommandResults:
+    """
+    :param client: Cisco Umbrella Client for the api request.
+    :param args: args from the user for the command.
+    """
+    page = args.get('page', '')
+    limit = args.get('limit', '')
+    suffix = domains_list_suffix(page=page, limit=limit)
+    response = client.get_domains_list(suffix=suffix)
+    domains_list = create_domain_list(client, response)
+    domains_list = domains_list[:int(limit)]
+    readable_output = tableToMarkdown(t=domains_list, name='List of Domains', headers=['Domain'])
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='CiscoUmbrellaEnforcement.Domains',
+        outputs_key_field='',
+        outputs=domains_list
+    )
+
+
+def domain_event_add_command(client: Client, args: dict) -> str:
+    """
+    :param client: Cisco Umbrella Client for the api request.
+    :param args: args from the user for the command.
+    :returns (str) confirmation or error regarding adding a new domain.
+    """
+    alert_time = args.get('alert_time')
+    device_id = args.get('device_id')
+    dst_domain = args.get('dst_domain')
+    dst_url = args.get('dst_url')
+    device_version = args.get('device_version')
+
+    new_event = {
+        "alertTime": alert_time,
+        "deviceId": device_id,
+        "deviceVersion": device_version,
+        "dstDomain": dst_domain,
+        "dstUrl": dst_url,
+        "eventTime": alert_time,
+        "protocolVersion": "1.0a",
+        "providerName": "Security Platform"
+    }
+
+    response = client.add_event_to_domain(new_event)
+    if response.get('id'):
+        ActionResult = f"New event was added successfully, The id is {str(response.get('id'))}."
+    else:
+        ActionResult = "New event's addition failed."
+    return ActionResult
+
+
+def domain_delete_command(client: Client, args: dict) -> str:
```

Comment:
As I've commented in the yml, let's add a key of `Deleted: True` to the deleted domains.